src/SDK/Resource/Detectors/Service.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Resource\Detectors;
+
+use OpenTelemetry\SDK\Common\Attribute\Attributes;
+use OpenTelemetry\SDK\Resource\ResourceDetectorInterface;
+use OpenTelemetry\SDK\Resource\ResourceInfo;
+use OpenTelemetry\SemConv\ResourceAttributes;
+
+/**
+ * @see https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#service-experimental
+ */
+final class Service implements ResourceDetectorInterface
+{
+    public function getResource(): ResourceInfo
+    {
+        static $serviceInstanceId = null;
+
+        if ($serviceInstanceId === null) {
+            $serviceInstanceId = \Ramsey\Uuid\Uuid::uuid4()->toString();
+        }
```

Comment:
```suggestion
        static $serviceInstanceId;
        $serviceInstanceId ??= Uuid::uuid4()->toString();
```
plus, a `use Ramsey\Uuid\Uuid;` up above